beam/core/src/main/java/cz/o2/proxima/beam/direct/io/BeamCommitLogReader.java

```diff
@@ -56,6 +56,7 @@
     private final DirectUnboundedSource source;
     @Getter private final BeamCommitLogReader reader;
 
+    private Instant previousWatermark = BoundedWindow.TIMESTAMP_MIN_VALUE;
```

Comment:
Maybe also here we should use `Watermarks.MIN_WATERMARK`

Reply:
this is a beam source, so we need a joda time instant here, so this is perfectly fine

Reply:
true! My mistake.